x-pack/plugins/ml/server/plugin.ts

```diff
@@ -158,25 +191,7 @@ export class MlServerPlugin
         () => this.dataViews
       ),
       mlLicense: this.mlLicense,
-    };
-
-    const resolveMlCapabilities = async (request: KibanaRequest) => {
-      if (this.capabilities === null) {
-        return null;
-      }
-      const capabilities = await this.capabilities.resolveCapabilities(request);
-      return capabilities.ml as MlCapabilities;
-    };
-
-    const getSpaces = plugins.spaces
-      ? () => coreSetup.getStartServices().then(([, { spaces }]) => spaces!)
-      : undefined;
-
-    const getDataViews = () => {
-      if (this.dataViews === null) {
-        throw Error('Data views plugin not initialized');
-      }
-      return this.dataViews;
+      sharedServicesProviders,
```

Comment:
as `sharedServicesProviders` is only needed for the alerting routes this can just be passed in as a second parameter to `alertingRoutes`
The pattern we have been using here is `routeInit`  only contains core items that are required by all routes.
Any additional services or functions are passed in as extra arguments to the route providers.

Reply:
Some time ago there was a discussion about providing the `alertingService` for other plugins so it can be used by other alerting rules. That's why we decided to make it a part of the `sharedServices`.

Reply:
yes, but it's not used by every route and so doesn't need to be in `routeInit`

Reply:
changed in https://github.com/elastic/kibana/pull/118006/commits/6ebc21bd2b37b510e121df8602cfed846048a003